parser.go

```diff
@@ -36,19 +36,26 @@ type Parser struct {
 	RSSTranslator  Translator
 	JSONTranslator Translator
 	UserAgent      string
+	AuthConfig     *Auth
 	Client         *http.Client
 	rp             *rss.Parser
 	ap             *atom.Parser
 	jp             *json.Parser
 }
 
+type Auth struct {
+	Username string
+	Password string
+}
+
 // NewParser creates a universal feed parser.
 func NewParser() *Parser {
 	fp := Parser{
-		rp:        &rss.Parser{},
-		ap:        &atom.Parser{},
-		jp:        &json.Parser{},
-		UserAgent: "Gofeed/1.0",
+		rp:         &rss.Parser{},
+		ap:         &atom.Parser{},
+		jp:         &json.Parser{},
+		UserAgent:  "Gofeed/1.0",
+		AuthConfig: &Auth{},
```

Comment:
```suggestion
```
equivalent to setting `AuthConfig: nil` manually.
Otherwise, the test l.111 will succeed and will add empty data to the header.